runtime/onert/core/src/util/shapeinf/Range.cc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util/ShapeInference.h"
+#include <cmath>
+namespace onert
+{
+namespace shape_inference
+{
+// helper function
+template <typename T> ir::Shape inferRangeShape(T start_val, T limit_val, T delta_val)
+{
+  ir::Shape out_shape(static_cast<int>(1));
+
+  out_shape.dim(0) =
+      (std::is_integral<T>::value
+           ? ((std::abs(*start_val - *limit_val) + std::abs(*delta_val) - 1) / std::abs(*delta_val))
+           : std::ceil(std::abs((*start_val - *limit_val) / *delta_val)));
+  return out_shape;
+}
+
+// StaticInferer at compilation time
+void StaticInferer::visit(const ir::operation::Range &op)
+{
+  const auto start_idx{op.getInputs().at(ir::operation::Range::Input::START)};
+  const auto limit_idx{op.getInputs().at(ir::operation::Range::Input::LIMIT)};
+  const auto delta_idx{op.getInputs().at(ir::operation::Range::Input::DELTA)};
+  const auto &start_op = _operands.at(start_idx);
+  const auto &limit_op = _operands.at(limit_idx);
+  const auto &delta_op = _operands.at(delta_idx);
+
+  // get mutable output operand
+  const auto output_idx = op.getOutputs().at(0);
+  ir::Operand &output = _operands.at(output_idx);
+  // if any input is dynamic, output also becomes dynamic
+  if (start_op.info().isDynamic() || limit_op.info().isDynamic() || delta_op.info().isDynamic())
+  {
+    output.info().setDynamic();
+    return;
+  }
+}
```

Comment:
I did not allocate output shape here mainly because the output shape is dependent on the input operand data, and not its shape. As this function is called in graph compilation stage (please correct me if wrong), any attempt to access operand data or tensor buffer will give zero value or `nullptr`. This is something I discussed under issue #1875.

The testcases with constant input operands were validated successfully. Unless I am missing out important testcases, the shape allocation seems redundant when input operands are constants. Again, please correct me if I am wrong.

Reply:
First of all, output Tensor is 1-D Tensor. I am not sure whether my opinion is right or not.
for example, if all of inputs are static, you can allocate output shape. In my opinion.

@hyunsik-yoon Please give your comment.

Reply:
IMHO, if all inputs are constant, the output shape can be calculated and set.

Reply:
I tried accessing the operand data (`operand.asScalar()`), and it does show zero values, which makes it challenging to set the output shape here. In any case, such a setting is redundant, as all inputs are constants.

Reply:
```suggestion
}
if (start_op.isConstant() && limit_op.isConstant() && delta_op.isConstant())
  {
    assert(start_op.typeInfo().type() == limit_op.typeInfo().type() && start_op.typeInfo().type() == delta_op.typeInfo().type());
    if (output.typeInfo().type() == ir::DataType::FLOAT32)
    {
      auto new_shape = inferRangeShape<float>(reinterpret_cast<const float *>(start_op.data()->base()),
                                              reinterpret_cast<const float *>(limit_op.data()->base()),
                                              reinterpret_cast<const float *>(delta_op.data()->base()));
      output.info().shape(new_shape);
    }
    else if (output.typeInfo().type() == ir::DataType::INT32)
    {
      auto new_shape = inferRangeShape<int32_t>(reinterpret_cast<const int32_t *>(start_op.data()->base()),
                                                reinterpret_cast<const int32_t *>(limit_op.data()->base()),
                                                reinterpret_cast<const int32_t *>(delta_op.data()->base()));
      output.info().shape(new_shape);
    }
  }
```

Reply:
Changing the prototype of `inferRangeShape` to `const T*` from `const T` gives the following build errors in use of `std::abs`:

```
error: invalid conversion from ‘int*’ to ‘long long int’ [-fpermissive]
   30 |            ? ((std::abs(*start_val - *limit_val) + std::abs(*delta_val) - 1) / std::abs(*delta_val))
      |                                                                                ~~~~~~~~^~~~~~~~~~~~
      |                                                                                        |
      |                                                                                        int*
```

Reply:
I have not yet tried it, because of the build errors explained in my reply below. However, I did try `asScalar()` earlier and it gave off 0 values. As I understand, `asScalar` returns `*(reinterpret_cast<const T *>(_data->base()))`. So I do not understand how the code snippet above would return valid values. Please correct me if my understanding is wrong.

Reply:
Now `inferRangeShape` use params as pointer, but the result of `asScalar` is not potiner.

Reply:
actually, I meant to point out the bigger issue, which is, `start_op->data()` is null. So regardless of `inferShape`, this will give an assert.

Reply:
> IMHO, if all inputs are constants, the output shape can be calculated and set.

@hyunsik-yoon  
I talked to @dr-venkman .
If all inputs are constants, output shape probably are the same shape as model. If it is right, setting output shape is redundant. 
Nonetheless, there is nothing wrong with setting output shape.

Reply:
@dr-venkman @ragmani 
I also guess so. 
To make sure, it would be better to add `assert(output.shape == inferRangeShape(...))`.

Reply:
@hyunsik-yoon 
This is similar to `Reshape` operation. I'm not sure which one of shape has higher priority if both of shape are different.

Reply:
@hyunsik-yoon 
This is similar to `Reshape` operation. I'm not sure which one of shape has higher priority if both of shape are different. 
Anyway, I agree on your opinion because in that case, this operation's inputs are constant. I will do it.

Reply:
@dr-venkman Do you agree on his opinion?

Reply:
I agree that checking `assert(output.shape == inferRangeShape(...))` is a good idea to check that TFLiteConverter does indeed set the output shape. If the shape do not match, then setting the shape here would be justified.